DeckPickerMenuOption Bug fix

nayeemdv opened the pull request:
## Pull Request template

## Purpose / Description
Added A dismiss() which closes DeckPickerMenuOption..

## Fixes
Fixes #13740

## How Has This Been Tested?
Tested on Redmi note 8 pro..

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

https://user-images.githubusercontent.com/91174350/234815622-f9bb84d4-79c4-4c4f-8a91-ec3f131dcbdb.mp4

criticalAY commented on AnkiDroid/src/main/java/com/ichi2/anki/dialogs/DeckPickerContextMenu.kt:
```diff
@@ -160,6 +160,7 @@ class DeckPickerContextMenu(private val collection: Collection) : AnalyticsDialo
                 (activity as AnkiActivity).dismissAllDialogFragments()
             }
         }
+        dismiss()
```
Thank you! Nice change and looks good but let's take the case where the user cancels the rename dialog or cancels creating a subdeck then in this case the user would have to reopen the menu to choose another option from the menu.
I don't have an issue if the other reviewers decide to go with it

nayeemdv replied:
You are right. And it actually creates some problems with the other options.
can i delete this pr from my side

criticalAY replied:
You don't have to apologize we all are just volunteers and I appreciate the time you gave for noticing this thing and digging into it. I would wait for another reviewer to respond before taking a final call

mikehardy replied:
I am not bothered, personally as a user, by having the full menu close if I cancel a sub-menu action.

mikehardy reviewed (approved):
This works for me. I may have a different personal opinion than others though.

Second reviewer please note the conversation above between @criticalAY and the new dev @nayeemdv (Hi 👋 😄 ) as you may feel differently?

david-allison reviewed (approved):
Better than we are now. Approved. Thank you!!!!!

I'd (mildly) request a `@NeedsTest("message")` as a follow-up, so we can define and verify behaviour.

criticalAY commented:
Ok then good to go, Thankyou!

david-allison commented:
And we're in! Thank you

nayeemdv commented:
This PR also closes all Upcoming FragmentDialogs...
So I added DismissAllDialogFragments only where it was needed and removed the overall dismiss.
This resolves all the options except "Custom Study Option".
what should i do next???
